pkg/compressor/compressor.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compressor
+
+import (
+	"compress/gzip"
+	"compress/lzw"
+	"compress/zlib"
+	"fmt"
+	"io"
+
+	"github.com/sirupsen/logrus"
+)
+
+// CompressSnapshot takes uncompressed data as input and compress the data according to Compression Policy
+// and write the compressed data into one end of pipe.
+func CompressSnapshot(data io.ReadCloser, compressionPolicy string) io.ReadCloser {
+	pReader, pWriter := io.Pipe()
+
+	var gWriter io.WriteCloser
+	logger := logrus.New().WithField("actor", "compressor")
+	logger.Infof("start compressing the snapshot using %v Compression Policy", compressionPolicy)
+
+	switch compressionPolicy {
```

Comment:
Can you please add the default case and return error for unknown compression policy (you will have to change the function signature)?

Reply:
actually unknown compression Policy get handled [here](https://github.com/gardener/etcd-backup-restore/pull/293/files#diff-7adcf3334be4957b8c200cd4ad81c23a84d4be71d629640291a51ba3b20a198eR38-R51) .... but still I can default case for safety purpose (it will be unreachable code.)

Reply:
Yet, there is a default case in the decompress case. Additional checks don't hurt.